internal/notifier/github.go

```diff
@@ -0,0 +1,120 @@
+/*
+Copyright 2020 The Flux CD contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package notifier
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/fluxcd/pkg/recorder"
+	"github.com/google/go-github/v32/github"
+	"golang.org/x/oauth2"
+)
+
+type GitHub struct {
+	Owner  string
+	Repo   string
+	Client *github.Client
+}
+
+func NewGitHub(addr string, token string) (*GitHub, error) {
+	owner, repo, err := parseGithubAddress(addr)
+	if err != nil {
+		return nil, err
+	}
+
+	ctx := context.Background()
+	ts := oauth2.StaticTokenSource(&oauth2.Token{AccessToken: token})
+	tc := oauth2.NewClient(ctx, ts)
+	client := github.NewClient(tc)
+
+	return &GitHub{
+		Owner:  owner,
+		Repo:   repo,
+		Client: client,
+	}, nil
+}
+
+// Post Github commit status
+func (g *GitHub) Post(event recorder.Event) error {
+	revString, ok := event.Metadata["revision"]
+	if !ok {
+		return errors.New("Missing revision metadata")
+	}
+	rev, err := parseRevision(revString)
+	if err != nil {
+		return err
+	}
+	state, err := toGitHubState(event.Severity)
+	if err != nil {
+		return err
+	}
+
+	desc := fmt.Sprintf("%v %v %v", event.InvolvedObject.Kind, event.InvolvedObject.Name, event.Reason)
```

Comment:
We need to split the reason into words and make them lower case, here is a function you can copy/paste https://github.com/fatih/camelcase/blob/master/camelcase.go#L47

Reply:
Right now the reason seems to be the same as the severity. I have only seen the values "info" and "error" but I will implement the function in case it changes.

Reply:
> Right now the reason seems to be the same as the severity

This is a bug, the reason should be one of: https://github.com/fluxcd/kustomize-controller/blob/master/api/v1alpha1/condition_types.go#L56-L90

Reply:
Fixed in https://github.com/fluxcd/kustomize-controller/pull/76